frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,151 @@
+import { createEffect, createSignal } from "solid-js";
+import { createStore } from "solid-js/store";
+import { styled } from "solid-styled-components";
+import Button from "../components/form/button";
+import InputEmail from "../components/form/email-input";
+import InputHash from "../components/form/hash-input";
+import Slider from "../components/slider";
+import InputFile from "../components/form/file-input";
+import { getFilesUrls } from "../helpers/getFilesUrl";
+import { validateEmail } from "../helpers/validation";
+
+const Train = () => {
+  // store of data from inputs of form
+  const [formData, setFormData] = createStore({
+    email: "",
+    files: [],
+  });
+
+  // store of errors inputs of form
+  const [formErrors, setFormErrors] = createStore({
+    email: "",
+    files: "",
+  });
+
+  const [hash, setHash] = createSignal("");
+
+  const [imagesList, setImagesList] = createSignal([]);
+
+  // set form data when inputs changed
+  const handleFormChange = (event) => {
+    const { target } = event;
+    setFormData(() => {
+      if (target.type === `file`) {
+        return { [target.name]: target.files };
+      } else {
+        return { [target.name]: target.value };
+      }
+    });
+  };
+
+  createEffect(() => {
+    const isValidEmail = validateEmail(formData.email);
+    const isValidFiles = formData.files.length > 10;
+
+    if (isValidEmail) {
+      setFormErrors({ email: "" });
+    }
+
+    if (isValidFiles) {
+      setFormErrors({ files: "" });
+    }
+  });
+
+  const handleFormSubmit = async () => {
+    const email = formData.email;
+    const isValidEmail = validateEmail(formData.email);
+    const isValidFiles = formData.files.length > 10;
+
+    // Create new FormData object and append files
+    const files = new FormData();
+    if (isValidEmail && isValidFiles) {
+      for (let i = 0; i < formData.files.length; i++) {
+        const currFile = formData.files[i];
+        files.append(`file-${i}`, currFile, currFile.name);
+      }
+
+      // Uploading the files using the fetch API to the server (temporarily with json for static)
+      const res = await fetch(
+        `${import.meta.env.VITE_API_URL}/scene-representation.json`,
+        {
+          method: "POST",
+          body: { files, email },
+        }
+      );
+      const hash = await res.json();
+      setHash(hash);
+    } else {
+      if (!isValidEmail) {
+        setFormErrors({ email: "Email is not valid" });
+      }
+      if (!isValidFiles) {
+        setFormErrors({ files: "Files must be more than 10" });
+      }
+    }
+  };
+
+  // set images urls when input file is updated
+  createEffect(() => {
+    setImagesList(getFilesUrls(formData.files));
+  });
```

Comment:
It is a bad practice to call setters in the `createEffect`. Please, read the [docs](https://www.solidjs.com/docs/latest/api#createeffect)